packages/relic/test/headers/new/special/accept_encoding_header_test.dart

```diff
@@ -0,0 +1,262 @@
+import 'package:relic/src/headers/headers.dart';
+import 'package:relic/src/relic_server.dart';
+import '../../headers_test_utils.dart';
+
+void main() {
+  group('Given an Accept-Encoding header with the strict flag true', () {
+    late RelicServer server;
+
+    setUp(() async {
+      try {
+        server = await RelicServer.bind(InternetAddress.loopbackIPv6, 0);
+      } on SocketException catch (_) {
+        server = await RelicServer.bind(InternetAddress.loopbackIPv4, 0);
+      }
+    });
+
+    tearDown(() => server.close());
+
+    test(
+      'when an empty Accept-Encoding header is passed then it should throw TestInvalidHeaderException',
+      () async {
+        expect(
+          () async => await getServerRequestHeaders(
+            server: server,
+            headers: {'accept-encoding': ''},
+            strictHeaders: true,
+          ),
+          throwsA(isA<TestInvalidHeaderException>()),
+        );
+      },
+    );
+
+    test(
+      'when an Accept-Encoding header with invalid quality values is passed then it should throw TestInvalidHeaderException',
+      () async {
+        expect(
+          () async => await getServerRequestHeaders(
+            server: server,
+            headers: {'accept-encoding': 'gzip;q=abc, deflate, br'},
+            strictHeaders: true,
+          ),
+          throwsA(isA<TestInvalidHeaderException>()),
+        );
+      },
+    );
+  });
+
+  group('Given an Accept-Encoding header with the strict flag false', () {
+    late RelicServer server;
+
+    setUp(() async {
+      try {
+        server = await RelicServer.bind(InternetAddress.loopbackIPv6, 0);
+      } on SocketException catch (_) {
+        server = await RelicServer.bind(InternetAddress.loopbackIPv4, 0);
+      }
+    });
+
+    tearDown(() => server.close());
+
+    test(
+      'when no Accept-Encoding header is passed then it should default to gzip',
+      () async {
+        Headers headers = await getServerRequestHeaders(
+          server: server,
+          headers: {},
+          strictHeaders: false,
+        );
+
+        expect(
+          headers.acceptEncoding?.encodings.map((e) => e.encoding).toList(),
+          equals(['gzip']),
+        );
+        expect(
+          headers.acceptEncoding?.encodings.map((e) => e.quality).toList(),
+          equals([1.0]),
+        );
+      },
+    );
+
+    group('when a single', () {
+      test(
+        'valid Accept-Encoding header is passed then it should parse correctly',
+        () async {
+          Headers headers = await getServerRequestHeaders(
+            server: server,
+            headers: {'accept-encoding': 'gzip'},
+            strictHeaders: false,
+          );
+
+          expect(
+            headers.acceptEncoding?.encodings.map((e) => e.encoding).toList(),
+            equals(['gzip']),
+          );
+          expect(
+            headers.acceptEncoding?.encodings.map((e) => e.quality).toList(),
+            equals([1.0]),
+          );
```

Comment:
Suggestion: Break this out into a separate test explaining that the default value is assigned if no quality is present.